src/pybind/mgr/dashboard/frontend/src/app/shared/components/progress/progress.component.ts

```diff
@@ -0,0 +1,21 @@
+import { Component, Input } from '@angular/core';
+import { Icons } from '../../enum/icons.enum';
+
+@Component({
+  selector: 'cd-progress',
+  templateUrl: './progress.component.html',
+  styleUrls: ['./progress.component.scss']
+})
+export class ProgressComponent {
+  icons = Icons;
+  @Input() progress: number;
+  @Input() description: string;
+  @Input() loading: boolean;
+  @Input() inProgress: boolean;
+  @Input() executingTaskName: string;
+  @Input() currentStep: string; // For upgradeStatus.which
+  @Input() servicesComplete: string; // For upgradeStatus.services_complete
+  @Input() progressText: string; // For upgradeStatus.progress
+  @Input() message: string; // For upgradeStatus.message
```

Comment:
I think these are some extra props that are required for upgrade scenario specifically so had to add them in this component. Regarding carbon am not sure if we are taking it up in this release and since I haven't worked on carbonization efforts yet, so not sure will be able to complete it in time.

Reply:
Lets not do uprade in this PR as it requires a lot of thinking and work. Thanks